core/src/main/scala/better/files/File.scala

```diff
@@ -573,7 +573,7 @@ class File private (val path: Path)(implicit val fileSystem: FileSystem = path.g
     inputStream(openOptions).apply(_.asObjectInputStream().deserialize[A])
 
   def register(service: WatchService, events: File.Events = File.Events.all): this.type = {
-    path.register(service, events.toArray)
+    path.register(service, events.toArray.map(_.asInstanceOf[WatchEvent.Kind[_] | Null]))
```

Comment:
**Pass array into java method**
Pass `Array[T]` as argument to java method that requires an array. Java arrays have type `Array[T | Null]`.